include/cvc5/cvc5_proof_rule.h

```diff
@@ -3773,6 +3773,20 @@ enum ENUM(ProofRewriteRule)
    * \endverbatim
    */
   EVALUE(STR_IN_RE_EVAL),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Strings -- regular expression membership evaluation with NFA**
+   *
+   * .. math::
+   *
+   *   \mathit{str.in\_re}(s, R) = c
+   *
+   * where :math:`s` is a constant string, :math:`R` is a constant regular
+   * expression and :math:`c` is true or false.
+   *
+   * \endverbatim
+   */
+  EVALUE(STR_IN_RE_EVAL_NFA),
```

Comment:
I think I'd prefer not (temporarily) changing the API with this change. 

What I'd recommend is: 
- We merge this PR with the new Eunoia implementation, but have it unable to be activated for now.
- I will test on my end the addition (to confirm the improvement) and then replace the current implementation with the new one.  Then, I will replace the Eunoia signature for str_in_re_eval with your implementation and send the PR that activates it.

Sound good?